src/iop/rawoverexposed.c

```diff
@@ -92,59 +92,21 @@ static void process_common_setup(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *
   // 4BAYER is not supported by this module yet anyway.
   const int ch = (dev->image_storage.flags & DT_IMAGE_4BAYER) ? 4 : 3;
 
-  float threshold;
-
-  // the clipping is detected as >1.0 after white level normalization
-
-  /*
-   * yes, technically, sensor clipping needs to be detected not accounting
-   * for white balance.
-   *
-   * but we are not after technical sensor clipping.
-   *
-   * pick some image that is overexposed, disable highlight clipping, apply
-   * negative exposure compensation. you'll see magenta highlight.
-   * if comment-out that ^ wb division, the module would not mark that
-   * area with magenta highlights as clipped, because technically
-   * the channels are not clipped, even though the colour is wrong.
-   *
-   * but we do want to see those magenta highlights marked...
-   */
-
-  if(piece->pipe->dsc.temperature.enabled)
-  {
-    threshold = FLT_MAX;
-
-    // so to detect the color clipping, we need to take white balance into account.
-    for(int k = 0; k < ch; k++) threshold = fminf(threshold, piece->pipe->dsc.temperature.coeffs[k]);
-  }
-  else
-  {
-    threshold = 1.0f;
-  }
-
-  threshold *= dev->rawoverexposed.threshold;
+  // the clipping is detected as (raw value > threshold) 
+  float threshold = dev->rawoverexposed.threshold;
 
   for(int k = 0; k < ch; k++)
   {
     // here is our threshold
     float chthr = threshold;
 
-    // but we check it on the raw input buffer, so we need backtransform threshold
-
-    // "undo" temperature iop
-    if(piece->pipe->dsc.temperature.enabled) chthr /= piece->pipe->dsc.temperature.coeffs[k];
-
     // "undo" rawprepare iop
     chthr *= piece->pipe->dsc.rawprepare.raw_white_point - piece->pipe->dsc.rawprepare.raw_black_level;
     chthr += piece->pipe->dsc.rawprepare.raw_black_level;
```

Comment:
This seems like it could trigger a bug if the user or camera defines multiple different black levels. Maybe worth to use the actual separate black levels instead of their averages.

`pipe->dsc.rawprepare.raw_black_level` is the average of the `separate black levels`.

Reply:
Note that I meant lines 104 and 105 which are unchanged by this PR, i.e. not a regression.
It is an existing bug if we want to classify it as such but could be a good time to fix it while we are at it.